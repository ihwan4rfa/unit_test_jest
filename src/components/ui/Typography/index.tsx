import React from "react";
import { TypographyProps } from "./types.type";

const Typography = (props: TypographyProps) => {
  const { children, type = "h1" } = props;

  const TypographyElement = type;
  return <TypographyElement>{children}</TypographyElement>;
};

export default Typography;
