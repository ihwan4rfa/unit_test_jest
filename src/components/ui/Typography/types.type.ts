export type TypographyProps = {
  children: React.ReactNode;
  type?: keyof JSX.IntrinsicElements;
};
