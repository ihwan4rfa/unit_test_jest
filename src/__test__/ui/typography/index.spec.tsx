import Typography from "@/components/ui/Typography";
import { render } from "@testing-library/react";

describe("typography component", () => {
  it("should render typography component", () => {
    const { container } = render(<Typography type="h1">test</Typography>);
    expect(container).toMatchSnapshot();
  });
});
