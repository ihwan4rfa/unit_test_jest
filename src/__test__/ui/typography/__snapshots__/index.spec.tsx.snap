// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography component should render typography component 1`] = `
<div>
  <h1>
    test
  </h1>
</div>
`;
